sdk/core/Azure.Core/src/Shared/OperationPollingStrategy.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Polling strategy of <see cref="OperationInternalBase"/>.
+    /// </summary>
+    internal abstract class OperationPollingStrategy
+    {
+        protected const string RetryAfterHeaderName = "Retry-After";
+        protected const string RetryAfterMsHeaderName = "retry-after-ms";
+        protected const string XRetryAfterMsHeaderName = "x-ms-retry-after-ms";
+
+        /// <summary>
+        /// Get the interval of next polling iteration.
+        /// </summary>
+        /// <remarks>Note that the value could change per call.</remarks>
+        /// <param name="response">Server response.</param>
+        /// <returns>Polling interval of next iteration.</returns>
+        public abstract TimeSpan GetNextWait(Response response);
+
+        protected static TimeSpan GetMaxIntervalFromResponseAndIntrinsic(Response response, TimeSpan intrinsic)
+        {
+            if (TryGetRetryAfter(response, out TimeSpan? retryAfter))
+            {
+                return GetMaxInterval(retryAfter!.Value, intrinsic);
+            }
+            return intrinsic;
+        }
+
+        protected static TimeSpan GetMaxInterval(TimeSpan a, TimeSpan b)
+        {
+            return a > b ? a : b;
+        }
+
+        protected static bool TryGetRetryAfter(Response response, out TimeSpan? retryAfter)
```

Comment:
I would move this into a RetryAfterPollingStrategy.  I want to separate the logic of the polling strategy from the choice of the which strategy to use.

The logic to use will do the following
If the response has retry-after use RetryAfterPollingStrategy
else if a strategy was passed in use that
else default to ConstantPollingStrategy